waiter/src/waiter/auth/oidc.clj

```diff
@@ -328,20 +358,28 @@
    request-handler]
   (let [oidc-authority (utils/uri-string->host oidc-authorize-uri)]
     (fn oidc-auth-handler [request]
-      (let [oidc-mode-delay (delay (request->oidc-mode allow-oidc-auth-api? allow-oidc-auth-services? oidc-default-mode request))]
-        (cond
-          (or (auth/request-authenticated? request)
-              (= :disabled @oidc-mode-delay)
-              ;; OIDC auth is no-op when request cannot be redirected
-              (not (supports-redirect? oidc-authority oidc-redirect-user-agent-products request))
-              ;; OIDC auth is avoided if client already has too many challenge cookies
-              (too-many-oidc-challenge-cookies? request oidc-num-challenge-cookies-allowed-in-request))
-          (request-handler request)
+      (-> (let [oidc-mode-delay (delay (request->oidc-mode allow-oidc-auth-api? allow-oidc-auth-services? oidc-default-mode request))
+                skip-oidc? (or (auth/request-authenticated? request)
+                               (= :disabled @oidc-mode-delay)
+                               ;; OIDC auth is no-op when request cannot be redirected
+                               (not (supports-redirect? oidc-authority oidc-redirect-user-agent-products request)))
+                many-oidc-challenge-cookies? (delay (too-many-oidc-challenge-cookies? request oidc-num-challenge-cookies-allowed-in-request))]
+            (cond
+              (or skip-oidc?
+                  ;; OIDC auth is avoided if client already has too many challenge cookies
+                  @many-oidc-challenge-cookies?)
+              (cond-> (request-handler request)
+                ;; update a 401 response to 403 if there were too many cookies and downstream auth options are disabled
```

Comment:
Probably good to document here _why_ we return 403 even though 401 is technically a better fit.

Reply:
Improved comment on why 403 is a more appropriate response.